### PR TITLE
[Deposit] Update UI to show pending assets

### LINKED ISSUE #1496: [Deposit] Update UI to show pending assets
- Show pending assets (if available) in a alert (warning) box at top of deposit component
- As long as we don have a "recover" tool has implemented, link to https://app.asgard.exchange/deposit-sym-recovery (just as temporary solution)
- Disable deposit form in case of pending assets

### PR DESCRIPTION
**Pull request by veado**
https://user-images.githubusercontent.com/61792675/121058303-7bd2e880-c7c0-11eb-8320-3f5e45fe3438.mp4

- [x] Show alert box in case of pending assets
- [x] Disable deposit form in case of pending assets
- [x] Reload data of pending assets by clicking "reload" button
- [x] Link to asgard.exchange to recover pending asset
- [x] Update i18n

Closes #1496

### CODE REVIEW

File: src/renderer/components/deposit/add/SymDeposit.tsx [hunk 1]
```diff
@@ -911,6 +917,47 @@ export const SymDeposit: React.FC<Props> = (props) => {
...
 
+  const hasPendingAssets: boolean = useMemo(
+    () =>
+      FP.pipe(
+        pendingAssetsRD,
+        RD.toOption,
+        O.map((pendingAssets): boolean => pendingAssets.length > 0),
+        O.getOrElse((): boolean => false)
+        // TODO @Veado Remove it, just for debugging
+        // () => true
```
Comment (veado) [inline src/renderer/components/deposit/add/SymDeposit.tsx]: Check your own TODO :sweat_smile:

File: src/renderer/components/deposit/add/Deposit.subcomponents.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    network,
+    loading
+  } = props
+  return (
+    <Styled.PendingAssetContainer>
+      <Styled.PendingAssetIcon asset={asset} network={network} />
+      <Styled.PendingAssetLabel asset={asset} />
+      <Styled.PendingAssetAmountLabel loading={loading}>
+        {formatAssetAmount({
+          amount: baseToAsset(amount),
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/deposit/add/Deposit.subcomponents.tsx:30]: should not we pass here `decimal` too ?

File: src/renderer/components/deposit/add/Deposit.subcomponents.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react'
```
Comment (veado) [inline src/renderer/components/deposit/add/Deposit.subcomponents.tsx:1]: Rename `Deposit.subcomponents` -> `DepositRecovery`

File: src/renderer/components/deposit/add/SymDeposit.tsx [hunk 4]
```diff
@@ -911,6 +917,47 @@ export const SymDeposit: React.FC<Props> = (props) => {
...
+      ),
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/deposit/add/SymDeposit.tsx]: nitpick
```ts
 FP.pipe(
        pendingAssetsRD,
        RD.toOption,
        O.chain(NEA.fromArray),
        O.getOrElse((): boolean => false)
      )
```

File: src/renderer/components/deposit/add/SymDeposit.tsx [hunk 5]
```diff
@@ -911,6 +917,47 @@ export const SymDeposit: React.FC<Props> = (props) => {
...
+
+    return FP.pipe(
+      pendingAssetsRD,
+      RD.fold(
+        () => <></>,
+        () => render(prevPendingAssets.current, true),
+        () => <></>,
+        (pendingAssets) => {
+          prevPendingAssets.current = pendingAssets
+          console.log('pendingAssets:', pendingAssets)
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/deposit/add/SymDeposit.tsx]: nitpick

File: src/renderer/i18n/ru/deposit.ts [hunk 6]
```diff
@@ -25,6 +25,12 @@ const deposit: DepositMessages = {
   'deposit.add.error.nobalances': 'Нет средств',
   'deposit.add.error.nobalance1': 'У вас нет средств на балансе {asset} в кошелке для вклада',
   'deposit.add.error.nobalance2': 'У вас нет средств на балансах {asset1} и {asset2} в кошельке для вклада',
+  'deposit.add.pendingAssets.title': 'Pending assets found - RU',
+  'deposit.add.pendingAssets.description':
+    'Following assets have been successfully sent, but the transaction of the other asset side has not been finalized or it was failed: - RU',
+  'deposit.add.pendingAssets.recoveryDescription':
+    'Please note: Transactions of an asset pair might take different time by running on different blockchains. In case of failures, you have a way to withdraw pending assets using the Recovery Tool of ASDGARDEX web at https://testnet.asgard.exchange/deposit-sym-recovery. This feature is currently not available in ASGARDEX desktop. - RU',
+  'deposit.add.pendingAssets.recoveryTitle': 'Open Recovery Tool - RU',
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/i18n/ru/deposit.ts]:
```
'deposit.add.pendingAssets.title': 'Обнаружены ожидающие ассеты', 
'deposit.add.pendingAssets.description':    'Эти ассеты были успешно отправлены, но транзакция для другого ассета не была завершена или возникла ошибка, при ее отправке:',
'deposit.add.pendingAssets.recoveryDescription':    'Обратите внимение: Транзакции для разных пар ассетов могут занимать разное время, так как запущены на разных блок-чейнах. В случае ошибки, у вас есть возможность изъять ожидающие ассеты с помощью утилиты возврата ASDGARDEX-веб на https://testnet.asgard.exchange/deposit-sym-recovery. Эта функция пока недоступна  в ПК версии ASGARDEX.',
'deposit.add.pendingAssets.recoveryTitle': 'Открыть утилиту для возврата',
```

File: src/renderer/services/thorchain/thorNode.ts [hunk 7]
```diff
@@ -143,13 +144,20 @@ const getLiquidityProviders = ({
...
-          A.map((provider) => ({
-            asset: provider.asset,
-            address: provider.rune_address,
-            assetAddress: provider.asset_address,
-            pendingRune: baseAmount(provider.pending_rune, THORCHAIN_DECIMAL),
-            pendingAsset: baseAmount(provider.pending_asset, assetDecimal)
-          }))
+          A.map((provider): LiquidityProvider => {
+            const pendingRuneAmount = baseAmount(provider.pending_rune, THORCHAIN_DECIMAL)
+            const pendingAssetAmount = baseAmount(provider.pending_asset, assetDecimal)
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/thorchain/thorNode.ts:149]: just realized that such transformation to the `BaseAmount` is a little bit incorrect... `assetDecimal` is not always the same as `selectedPoolAsset`'s decimal in general.
for the future i believe we need to pass pending amounts as strings directly and transform to the `BaseAmount` only "on demand" (at the view level when we know assets' decimals...)

File: src/renderer/services/thorchain/types.ts [hunk 8]
```diff
@@ -84,16 +84,19 @@ export type GetLiquidityProvidersParams = {
 }
 
 export type GetLiquidityProviderParams = GetLiquidityProvidersParams & {
-  runeAddress: O.Option<Address>
-  assetAddress: O.Option<Address>
+  runeAddress: Address
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/thorchain/types.ts:87]: i would keep `runeAddress: O.Option<Address>`:
- `assetAddress` is always required for both `sym`/`asym` providers
- `runeAddress` is optional to detect if we want to get `sym`/`asym` provider.
i know we now "ignore" all asym-related stuff but if this is a brand-new stuff we should be "aimed" to the future stuff

File: src/renderer/views/deposit/DepositView.tsx [hunk 9]
```diff
@@ -149,42 +158,26 @@ export const DepositView: React.FC<Props> = () => {
...
+      addressByChain$(THORChain),
+      assetWithDecimalLD
+    ]).pipe(
+      RxOp.switchMap(([network, oAssetAddress, oRuneAddress, assetWithDecimalRD]) => {
+        return FP.pipe(
+          sequenceTOption(oRuneAddress, oAssetAddress, RD.toOption(assetWithDecimalRD)),
+          O.fold(
+            (): LiquidityProviderLD => Rx.of(RD.initial),
+            ([runeAddress, assetAddress, assetWithDecimal]) =>
+              getLiquidityProvider({ assetWithDecimal, network, runeAddress, assetAddress })
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/views/deposit/DepositView.tsx:175]: i would move this data getter directly inside of `SymDepositView`. with optional `runeAddress` at the `getLiquidityProvider` we can easily split this logic between 2 views without passing-down `liquidityProvider` through few levels:
- `SymDepositView`: `getLiquidityProvider({ assetWithDecimal: ..., network: ..., runeAddress: O.some(Address), assetAddress: Address })`
- `AsymDepositView`: `getLiquidityProvider({ assetWithDecimal: ..., network: ..., runeAddress: O.none, assetAddress: Address })`

what do you think ?

**Review by thatStrangeGuyThorchain — CHANGES_REQUESTED** [review]

File: src/renderer/services/thorchain/thorNode.ts (hunk 7 above)
Comment (veado) [reply src/renderer/services/thorchain/thorNode.ts:149]: Good point!

File: src/renderer/components/deposit/add/Deposit.subcomponents.tsx (hunk 2 above)
Comment (veado) [reply src/renderer/components/deposit/add/Deposit.subcomponents.tsx:30]: `amount: BaseAmount` has already correct decimal included.

File: src/renderer/components/deposit/add/Deposit.subcomponents.tsx (hunk 3 above)
Comment (veado) [reply src/renderer/components/deposit/add/Deposit.subcomponents.tsx:1]: Will be another PR.

File: src/renderer/services/thorchain/types.ts (hunk 8 above)
Comment (veado) [reply src/renderer/services/thorchain/types.ts:87]: Good points as well!

File: src/renderer/views/deposit/DepositView.tsx (hunk 9 above)
Comment (veado) [reply src/renderer/views/deposit/DepositView.tsx:175]: Sounds good.

File: src/renderer/services/thorchain/thorNode.ts (hunk 7 above)
Comment (veado) [reply src/renderer/services/thorchain/thorNode.ts:149]: _Will be done with another PR - see #1512_

File: src/renderer/services/thorchain/types.ts (hunk 8 above)
Comment (veado) [reply src/renderer/services/thorchain/types.ts:87]: ~~_Will be done with another PR - see #1512_~~

File: src/renderer/views/deposit/DepositView.tsx (hunk 9 above)
Comment (veado) [reply src/renderer/views/deposit/DepositView.tsx:175]: ~~_Will be done with another PR - see #1512_~~

**Comment by veado** [comment]
_Merging now to have it in next release..._ All open suggestion will be done with another PR - see #1512

File: src/renderer/services/thorchain/types.ts (hunk 8 above)
Comment (veado) [reply src/renderer/services/thorchain/types.ts:87]: > runeAddress is optional to detect if we want to get sym/asym provider.

That's not correct, but as follow: "if member is assym - they will only have either rune_address or asset_address in their member details." (see https://thorchain.slack.com/archives/C012CLYR1DG/p1612347969232600?thread_ts=1612288793.231200&cid=C012CLYR1DG). We will check always both addresses then.

Check for pending asset is for sym. deposit needed only. If an asym. deposit might fail, there won't be a pending asset.

File: src/renderer/views/deposit/DepositView.tsx (hunk 9 above)
Comment (veado) [reply src/renderer/views/deposit/DepositView.tsx:175]: I'm going to skip this for same reason mentioned in https://github.com/thorchain/asgardex-electron/pull/1505#discussion_r647415243

**Pull request merged**